GVFS/GVFS.Common/PlaceholderListDatabase.cs

```diff
@@ -210,12 +216,12 @@ public void GetAllEntriesAndPrepToWriteAllEntries(out List<PlaceholderData> file
             }
         }
 
-        public Dictionary<string, PlaceholderListDatabase.PlaceholderData> GetAllFileEntries()
+        public HashSet<string> GetAllFilePaths()
         {
             try
             {
-                Dictionary<string, PlaceholderListDatabase.PlaceholderData> filePlaceholdersFromDiskByPath =
-                    new Dictionary<string, PlaceholderListDatabase.PlaceholderData>(Math.Max(1, this.EstimatedCount), StringComparer.Ordinal);
+                HashSet<string> filePlaceholdersFromDiskByPath =

```

Comment:
Minor: I think the "ByPath" part of the `filePlaceholdersFromDiskByPath` name was because this was a dictionary.  We could probably rename this to something simpler like `filePlaceholderPaths`